samples/js/photos/src/photo-grid-item.js

```diff
@@ -24,14 +29,17 @@ type Props = {
   gridWidth: number,
   nav: Nav,
   photo: Photo,
-  url: string,
   viewport: Viewport,
 };
 
 type State = {
-  size: PhotoSize,
-  sizeIsBest: boolean,
-  url: string,
+  // The NomsBlob we are currently supposed to be displaying. When this changes, we start loading
+  // it asynchronously and eventually change |url| to match.
+  blob: Ref<NomsBlob> | null,
+
+  // A blob: URL referring to a DOM blob we are displaying. This is populated once |blob| has
+  // loaded.
```

Comment:
The 2nd sentence is potentially misleading, because it looks like if the props change to a different blob, the url will stay the same as the original blob until the new one has loaded. Perhaps say "... else null or the previous loaded blob".

Speaking of which I wonder if this should just be string type and empty-string if there is no blob, up to you.